dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Resources;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Rest;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Skills;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+/// <summary>
+/// Class for extensions methods for <see cref="IKernel"/> interface.
+/// </summary>
+public static class KernelOpenApiExtensions
+{
+
+    /// <summary>
+    /// Imports OpenAPI document from a URL.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="url">Url to in which to retrieve the OpenAPI definition.</param>
+    /// <param name="httpClient">Optional HttpClient to use for the request.</param>
+    /// <param name="authCallback">Optional callback for adding auth data to the API requests.</param>
+    /// <returns>A list of all the semantic functions representing the skill.</returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportOpenApiSkillFromUrlAsync(
+        this IKernel kernel, string skillName, Uri url, HttpClient? httpClient = null, Func<HttpRequestMessage, Task>? authCallback = null)
+    {
+        Verify.ValidSkillName(skillName);
+
+        HttpResponseMessage? response = null;
+        try
+        {
+            if (httpClient == null)
+            {
+                // TODO Fix this:  throwing "The inner handler has not been assigned"
+                //using DefaultHttpRetryHandler retryHandler = new DefaultHttpRetryHandler(
```

Comment:
```suggestion
using var retryHandler = (new DefaultHttpRetryHandlerFactory(new HttpRetryConfig { MaxRetryCount = 3 })).Create(kernel.Log);
retryHandler.InnerHandler = new HttpClientHandler();
using HttpClient client = new(retryHandler);
```

Reply:
Or use the `kernel.Config.HttpHandlerFactory`

Reply:
(Fine to ignore, just for future updates)

Reply:
Thanks Lee, it will be added in one of the next iterations in scope of the retry handler task.